File: StatusIncrease.cpp
//
//  StatusIncrease.cpp
//  
//
//  Created by Muhammad Sohibul Maromi on 3/15/13.
//
//

#include "StatusIncrease.h"

StatusIncrease::StatusIncrease(int x, string s, int mp, int hp) : Item(100,"statusincrease") {
    mpMaxAddition = mp;
    hpMaxAddition = hp;
}

StatusIncrease::StatusIncrease(int mpp,int hpp) : Item(100,"statusincrease"){
	this->mpMaxAddition = mpp;
	this->hpMaxAddition = hpp;
}

StatusIncrease::StatusIncrease(const StatusIncrease& s) {
    mpMaxAddition = s.mpMaxAddition;
    hpMaxAddition = s.hpMaxAddition;
	
	harga = s.harga;
	itemName = s.itemName;
}

StatusIncrease::~StatusIncrease() {
    
}

StatusIncrease& StatusIncrease::operator=(const StatusIncrease& s) {
    mpMaxAddition = s.mpMaxAddition;
    hpMaxAddition = s.hpMaxAddition;
	
	harga = s.harga;
	itemName = s.itemName;
    
    return *this;
}

void StatusIncrease::execute(Monster& m) {
    m.setHP(m.getHP() + mpMaxAddition);
    m.setMP(m.getMP() + hpMaxAddition);
}

File: Android/model/StatusIncrease.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package model;

/**
 *
 * @author msmaromi
 */
public class StatusIncrease extends Item {
  private int mpMaxTambah;
  private int hpMaxTambah;
  
  public StatusIncrease(int harga, String nama, int mp, int hp) {
    super(harga, nama);
    mpMaxTambah = mp;
    hpMaxTambah = hp;
  }
  
  public StatusIncrease(StatusIncrease st) {
    super(st);
    mpMaxTambah = st.mpMaxTambah;
    hpMaxTambah = st.hpMaxTambah;            
  }
  
  public int getMPmaxTambah() {
    return mpMaxTambah;
  }
  
  public int getHPmaxTambah() {
    return hpMaxTambah;
  }
  
  public void setMPmaxTambah(int mp) {
    mpMaxTambah = mp;
  }
  
  public void setHPmaxTambah(int hp) {
    hpMaxTambah = hp;
  }
  
  @Override
  public void execute(Monster m) {
//    m.setHP(m.getHP()+hpMaxTambah);
//    m.setMP(m.getMP()+mpMaxTambah);
  }
}
